src/text/non_blocking/raw_text_reader.rs

```diff
@@ -343,7 +345,12 @@ impl<A: AsRef<[u8]>> RawTextReader<A> {
                     self.process_stream_item(item)
                 } else {
                     // If we are not at the end of the stream, we need to get more data.
-                    incomplete_text_error(self.buffer.lines_loaded(), self.buffer.line_offset())
+                    incomplete_text_error(
+                        "text",
+                        self.buffer.bytes_consumed(),
+                        self.buffer.lines_loaded(),
+                        self.buffer.line_offset(),
```

Comment:
A thought you can take or leave: maybe it makes sense to give `TextBuffer` a convenience method since it's the owner of all of these bits of information?

```rust
text_buffer.incomplete_text_error("text") // or something?
```

Reply:
I had considered something similar but didn't feel like the text buffer was the place to generate the error, since the reader was the one who really identified it.

In the latest commit I added a `get_position` function to `TextBuffer` and have it return a `Position` with all of that data. I also updated `incomplete_text_error*` to take a Position instead of individual values, which simplifies this a little more and encapsulates that positioning info a little better.